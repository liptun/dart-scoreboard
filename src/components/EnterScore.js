import React, { useState } from 'react'
import { connect } from 'react-redux'
import dartHit from '../libs/dartHit'
import { playerHit } from '../actions'




import '../styles/enter-score.scss'

const EnterScore = (props) => {
    const [score, setScore] = useState('' )

    const onScoreSubmit = (e) => {
        e.preventDefault()
        if (score.trim()) {
            const newHit = dartHit(score)
            props.dispatch(playerHit({ score: newHit }))
            setScore('')
        }
    }

    return (
        <div className="enter-score">
            <form onSubmit={onScoreSubmit}>
                <input
                    type="text"
                    value={score}
                    onChange={(e) => setScore(e.target.value)}
                />
                <button>Score</button>
            </form>
            <p>{dartHit(score)}</p>
        </div>
    )
}

const mapStateToProps = (state) => ({
    game: state.game,
})
export default connect(mapStateToProps)(EnterScore)
