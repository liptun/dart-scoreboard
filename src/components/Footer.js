import React from 'react'

import '../styles/footer.scss'

const Footer = () => (
    <footer className="footer">
        <p>
            Developed by{' '}
            <a href="https://github.com/liptun" target="_blank" rel="noreferrer">
                Rafa≈Ç Karczmarzyk aka LiptuN
            </a>
        </p>
    </footer>
)
export default Footer
