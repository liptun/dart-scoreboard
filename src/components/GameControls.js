import React from 'react'
import { connect } from 'react-redux'
import { endGame } from '../actions'

const GameControls = (props) => {
    const onClickUndo = () => {}
    const onClickReset = () => {}
    const onClickEnd = () => {
        props.dispatch(endGame())
    }

    return (
        <div>
            <button onClick={onClickUndo}>Undo</button>
            <button onClick={onClickReset}>Reset</button>
            <button onClick={onClickEnd}></button>
        </div>
    )
}

const mapStateToProps = (state) => ({
    game: state.game,
})
export default connect(mapStateToProps)(GameControls)
