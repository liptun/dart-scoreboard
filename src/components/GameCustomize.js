import React from 'react'
import { connect } from 'react-redux'
import { setGameType } from '../actions'

import '../styles/game-customize.scss'

const GameCustomize = (props) => {
    const onChangeGameType = (gameType) => {
        props.dispatch(setGameType({ gameType }))
    }

    return (
        <div className="game-customize">
            <select
                value={props.game.gameType}
                onChange={(e) => onChangeGameType(e.target.value)}
            >
                <option value={101}>101</option>
                <option value={301}>301</option>
                <option value={501}>501</option>
                <option value={701}>701</option>
                <option value={901}>901</option>
            </select>
        </div>
    )
}

const mapStateToProps = (state) => ({
    game: state.game,
})

export default connect(mapStateToProps)(GameCustomize)
