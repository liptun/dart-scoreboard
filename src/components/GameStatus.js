import React from 'react'
import { connect } from 'react-redux'
import randomThrowsMessage from '../libs/randomThrowsMessage'

import '../styles/game-status.scss'

const GameStatus = (props) => {
    const playerIndex = props.game.currentPlayerIndex
    const { name: playerName } = props.game.players[playerIndex]
    return (
        <div className="game-status">
            {props.game.winner ? (
                <p>
                    {props.game.players[props.game.currentPlayerIndex].name}{' '}
                    wins
                </p>
            ) : (
                <p>{randomThrowsMessage(playerName)}</p>
            )}
        </div>
    )
}

const mapStateToProps = (state) => ({
    game: state.game,
})
export default connect(mapStateToProps)(GameStatus)
