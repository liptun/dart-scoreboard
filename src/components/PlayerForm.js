import React, { useState } from 'react'
import { connect } from 'react-redux'
import { addPlayer } from '../actions'

import '../styles/player-form.scss'

const PlayerForm = (props) => {
    const [playerName, setPlayerName] = useState(
        `Player ${props.players.length + 1}`
    )

    const onFormSubmit = (e) => {
        e.preventDefault()
        if (playerName.trim()) {
            props.dispatch(addPlayer({ name: playerName }))
            setPlayerName(`Player ${props.players.length + 2}`)
        }
    }

    return (
        <div className="player-form">
            <form onSubmit={onFormSubmit}>
                <input
                    type="text"
                    placeholder="Player name"
                    value={playerName}
                    onChange={(e) => setPlayerName(e.target.value)}
                    onFocus={(e) => e.target.select()}
                />
                <button disabled={props.players.length >= 6}>Add</button>
            </form>
        </div>
    )
}
PlayerForm.defaultProps = {
    players: [],
}
export { PlayerForm }

const mapStateToProps = (state) => ({
    players: state.game.players,
})
export default connect(mapStateToProps)(PlayerForm)
