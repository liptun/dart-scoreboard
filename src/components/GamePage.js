import React from 'react'
import { connect } from 'react-redux'
import Scoreboard from './Scoreboard'
import EnterScore from './EnterScore'

import '../styles/game-page.scss'

const GamePage = (props) => {
    return (
        <div className="game-page">
            <h1>Game {props.game.gameType}</h1>
            <EnterScore />
            <Scoreboard />
        </div>
    )
}

const mapStateToProps = (state) => ({
    game: state.game
})
export default connect(mapStateToProps)(GamePage)
