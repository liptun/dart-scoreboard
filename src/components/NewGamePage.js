import React from 'react'
import { connect } from 'react-redux'
import PlayerList from './PlayerList'
import PlayerForm from './PlayerForm'
import GameCustomize from './GameCustomize'
import { startGame } from '../actions'
import '../styles/new-game-page.scss'

const NewGamePage = (props) => {
    const startNewGame = () => {
        props.history.push('/game')
        props.dispatch(startGame())
    }

    return (
        <div className="new-game-page">
            <h1>Dart 101</h1>
            <GameCustomize />
            <PlayerList />
            <PlayerForm />
            <button
                disabled={!props.game.players.length}
                className="btn btn-cta"
                onClick={() => startNewGame()}
            >
                <span>Start </span>
                <i className="ico ico-target"></i>
            </button>
        </div>
    )
}

const mapStateToProps = (state) => ({
    game: state.game,
})

export default connect(mapStateToProps)(NewGamePage)
