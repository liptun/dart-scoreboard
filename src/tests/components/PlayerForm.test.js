import React from 'react'
import { shallow } from 'enzyme'
import { PlayerForm } from '../../components/PlayerForm'

test('Render PlayerForm component', () => {
    const wrapper = shallow(<PlayerForm />)
    expect(wrapper).toMatchSnapshot()
})

// test('PlayerForm add player', () => {
//     const value = 'LiptuN'
//     const wrapper = shallow(<PlayerForm dispatch={() => {}} />)
//     wrapper.find('input').simulate('change', { target: { value } })
//     expect(wrapper.state('playerName')).toBe(value)
// })
