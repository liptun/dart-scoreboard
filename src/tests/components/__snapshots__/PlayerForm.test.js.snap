// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PlayerForm component 1`] = `
<div
  className="player-form"
>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Player name"
      type="text"
      value="Player 1"
    />
    <button
      disabled={false}
    >
      Add
    </button>
  </form>
</div>
`;
